tutorials/continuous-deployment-pipeline-google-container-engine-with-codeship/index.md

```diff
@@ -0,0 +1,275 @@
+1.  `CONTAINER_CLUSTER`: Give your container cluster a name, in this case use
+    `hello-express-cluster`
+1.  `GOOGLE_PROJECT_ID`: The `Project ID` is found in the [Google Cloud Engine dashboard](https://console.cloud.google.com/home/dashboard)
+1.  `GOOGLE_AUTH_EMAIL`: Use the `Service Account ID` for the [service account](https://console.cloud.google.com/iam-admin/serviceaccounts)
+    created in the last step.
+1.  `GOOGLE_AUTH_JSON`: Add the credentials downloaded in the previous step
+    here. You must replace the newlines with spaces by running
+    `tr '\n' ' ' < your_file_name`.
+
+Save the `.env` file once these items are finished.
+
+### Step 3: Run initial deployment
+
+You need to push an image and create your clusters in GKE initially before you
+can set up a fully automated pipeline in Codeship. In the `hello-express` source
+code is a `bin` folder with `bash` scripts perform these tasks using the
+Codeship Jet CLI.
+
+1.  Make sure the scripts in the `bin` folder are executable. In a terminal
+    window, navigate to the project folder and run the following command:
+
+        chmod +x ./bin/**
+
+1.  Update the `image` line in the `codeship-service.yml` file with your Google
+    Cloud Platform project ID.
+
+        app:
+          build:
+            dockerfile: Dockerfile
+            image: gcr.io/YOUR_PROJECT_IDE/hello-express #update this line using your Google Cloud Platform project ID
+        ...
+
+1.  Update the `image_name` line in the `codeship_steps.yml` file with your
+    Google Cloud Platform project ID.
+
+        - name: build-image
+          service: app
+          command: echo "Build completed"
+        - name: push-image-with-sha
+          service: app
+          type: push
+          image_name: "gcr.io/YOUR_PROJECT_ID/hello-express" #update this line using your Google Cloud Platform project ID
+          image_tag: "{{printf \"%.8s\" .CommitID}}"
+          registry: https://gcr.io
+          dockercfg_service: codeship_gcr_dockercfg
+        - name: tag-as-master
+          service: app
+          type: push
+          tag: master
+          image_name: "gcr.io/YOUR_PROJECT_ID/hello-express" #update this line using your Google Cloud Platform project ID
+          image_tag: "master"
+          registry: https://gcr.io
+          dockercfg_service: codeship_gcr_dockercfg
+        - name: gke-initial-deployment
+          service: codeship_gce_service
+          tag: master
+          command: bin/deploy --create
+        ...
+
+This pipeline will build the Docker image, and then push to Google Container
+Registry with a tag using the first 8 characters of the commit SHA for every
+commit to the repository. The third and fourth step will run only if the branch
+is tagged as master.
+
+You will run this pipeline locally using the [Codeship Jet CLI](https://documentation.codeship.com/pro/builds-and-configuration/cli/). There
+will be no commit SHA from git, but you will pass in a string using a flag.
```

Comment:
I don't quite follow this sentence. Is this telling me that I should take the value of the Container Registry tag and pass it in the `--ci-commit-id` flag? If not, what's it trying to say?

Reply:
Here is a suggested clarification - let me know if this helps clear things up.

>You will run this pipeline locally using the [Codeship Jet CLI](https://documentation.codeship.com/pro/builds-and-configuration/cli/). Since there is no git commit or branch to reference, use the `ci-commit-id` and `tag` flags with the Codeship Jet CLI to pass in values at runtime:

Reply:
The clarified version is much better! However, it's still not obvious to me where I would get the values for those flags. I think you should make that explicit.